src/bin/rustfmt.rs

```diff
@@ -7,23 +7,138 @@
 // <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
-
+#![feature(path_ext)]
+#![feature(rustc_private)]
 #![cfg(not(test))]
 
+#[macro_use]
+extern crate log;
 extern crate rustfmt;
+extern crate toml;
 
 use rustfmt::{WriteMode, run};
+use rustfmt::config::Config;
+
+
+use std::env;
+use std::fmt::{self, Write};
+use std::fs::{File, PathExt};
+use std::io::{self, Read};
+use std::path::{Path, PathBuf};
+
+use toml::Parser;
+
+// Error of the read_toml function.
+#[derive(Debug)]
+enum ReadError {
+    FmtError(fmt::Error),
+    IoError(io::Error),
+    TOMLError(String),
+}
+
+impl From<fmt::Error> for ReadError {
+    fn from(err: fmt::Error) -> ReadError {
+        ReadError::FmtError(err)
+    }
+}
+
+impl From<io::Error> for ReadError {
+    fn from(err: io::Error) -> ReadError {
+        ReadError::IoError(err)
+    }
+}
+
+impl fmt::Display for ReadError {
+
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            ReadError::FmtError(ref err) => {
+                write!(f, "{}", err)
+            }
+            ReadError::IoError(ref err) => {
+                write!(f, "{}", err)
+            }
+            ReadError::TOMLError(ref errmsg) => {
+                write!(f, "{}", errmsg)
+            }
+        }
+    }
+
+}
+
+// Tries to read a file and parse the contents as a toml table. Panics on error.
+fn read_toml<A: AsRef<Path>>(path: A) -> Result<toml::Table, ReadError> {
+    let mut file = try!(File::open(&path));
+    let mut contents = String::new();
+    try!(file.read_to_string(&mut contents));
+
+    let mut parser = Parser::new(&contents);
+    match parser.parse() {
+        Some(t) => Ok(t),
+        None => {
+            let mut errmsg = String::new();
+
+            try!(writeln!(errmsg, "TOML error in file {}", path.as_ref().display()));
```

Comment:
The `try!` here is misleading, since the `writeln!` will never fail:  `let mut errmsg = format!("TOML error in file {}\n", path.as_ref().display());`